lib/lhm/progress.rb

```diff
@@ -0,0 +1,202 @@
+require "lhm/printer"
+require "json"
+
+module Lhm
+  STATE_COPYING = "copying"
+  STATE_COPYING_DONE = "copying_done"
+  STATE_COPYING_FAILED = "copying_failed"
+  STATE_INITIAL = "initial"
+  STATE_SETUP_FAILED = "setup_failed"
+  STATE_SETUP_DONE = "setup_done"
+  STATE_SWITCHING_TABLES = "switching_tables"
+  STATE_SWITCHED_TABLES = "switched_tables"
+  STATE_SWITCHING_TABLES_FAILED = "switching_tables_failed"
+  STATE_TRIGGERS_DROPPED = "trigggers_dropeed"
```

Comment:
I identified the states based on the failure points in Lhm, so that's why there are many states. Also even if someone is not polling these many states, I don't see much downside in having more number of states that we have anticipated as these states are just of a `Hash` and don't really trigger any extra behaviour within the library.

Reply:
> The other thing is that there's no "state machine" per say. It's more like status changes, but that's just semantics I guess
True. My wording is a bit misleading as there isn't a real state machine here.